File: example_bot.c
// Example C bot using the Chess API

#include "chessapi.h"
#include <stdlib.h>

int main(int argc, char *argv[]) {
    for (int i = 0; i < 500; i++) {
        // at the start of our turn, get the board
        Board *board = chess_get_board();
        // find all legal moves for the position
        int len_moves;
        Move *moves = chess_get_legal_moves(board, &len_moves);
        // play one at random
        chess_push(moves[rand() % len_moves]);
        // cleanup
        free(moves);
        chess_free_board(board);
        // end our turn
        chess_done();
    }
}

File: example_bot.py
# Example Python bot using the Chess API

from pychessapi import API
from random import randrange

for i in range(500):
    # get the current board
    board = API.get_board()
    # get all legal moves for the position
    moves = board.get_legal_moves()
    # play one at random
    API.push(moves[randrange(0, len(moves))])
    # end our turn
    API.done()